pandas/core/groupby/generic.py

```diff
@@ -1884,11 +1884,9 @@ def func(df):
         numeric_only_default="False",
     )
     def idxmin(
-        self,
-        axis: Axis = 0,
-        skipna: bool = True,
-        numeric_only: bool = False,
+        self, axis: Axis = 0, skipna: bool = True, numeric_only: bool = False, **kwargs
```

Comment:
I don' think we should be adding kwargs to the function signature. Whenever `axis != self.axis` here we will raise, so I'd suggest instead just ignoring the `axis` argument altogether (but still leaving it in the method signature for now). 

Assuming this sounds like a good route, can you change the axis argument in the docstring to state that it is ignored. Right now we're using the shared_docs; I'd recommend copying this and adding it as an explicit docstring to this method - that would then allow us in the future to add groupby-specific docs for this method and correct some difference in behavior between DataFrame and DataFrameGroupBy.

Reply:
Thank you for the feedback, it does sound like a good route. As for the `axis` parameter, it does not seem intuitive to raise when we are stating in the docs that the argument will be ignored. We could however, instead of ignoring `axis`, just mention in the docs that `axis` and `self.axis` should match. If they don't match, we will raise with the message - they should match.

Reply:
> As for the `axis` parameter, it does not seem intuitive to raise when we are stating in the docs that the argument will be ignored.

I think this is in reply to my sentence `Whenever self.axis=1 and axis = 0 in idxmin we will raise (whether using with transform or not), so I'd suggest...`. When I said "we will raise", I mean "currently pandas raises". In other words, I think it's okay to ignore the `axis` argument because currently no one can be using `.groupby(..., axis=1).idxmin(axis=0)`.

> We could however, instead of ignoring axis, just mention in the docs that axis and self.axis should match. If they don't match, we will raise with the message - they should match.

I agree this is a viable route, but I think a bit odd. This is perhaps an over simplification, but it's like having a function

```
def foo(x=1):
    if x != 2:
        raise ValueError("You must set x to be 2")
    ...
```

Reply:
> When I said "we will raise", I mean "currently pandas raises". In other words, I think it's okay to ignore the axis argument because currently no one can be using .groupby(..., axis=1).idxmin(axis=0).

Okay now I understand what you mean. For now we will ignore `axis`, in the future remove it and directly take `axis` from the grouper.